Phase08/Phase05/Search/InvertedIndex.cs

```diff
@@ -0,0 +1,77 @@
+using Phase05.Utils;
+using System.Collections.Generic;
+using Phase05.DataSet;
+using System.Linq;
+
+namespace Phase05.Search
+{
+    public class InvertedIndex
+    {
+        private InvertedIndexContext Context;
+
+        public InvertedIndex(InvertedIndexContext context)
+        {
+            this.Context = context;
+        }
+
+        public InvertedIndex() { }
+
+        public void CreateIndex(List<string> documents)
+        {
+            for (int i = 0; i < documents.Count; i++)
+            {
+                var docId = AddDocument(documents[i]);
+                var tokenizedDoc = Tokenizer.Tokenize(documents[i]);
+                var docWords = Tokenizer.SplitDocument(tokenizedDoc);
+                foreach (string word in docWords)
+                    if (!word.Equals(""))
+                        AddToIndex(word, docId);
```

Comment:
https://stackoverflow.com/questions/1930982/when-should-i-call-savechanges-when-creating-1000s-of-entity-framework-object